e2e/testcases/oci_sync_test.go

```diff
@@ -35,56 +36,74 @@ import (
-	// which includes resources for tenant-a, tenant-b, tenant-c, tenant-d.
-	// Each tenant includes a NetworkPolicy, a Role and a RoleBinding.
-	// Image structure:
-	// .
-	//├── base
-	//│   ├── kustomization.yaml
-	//│   ├── namespace.yaml
-	//│   ├── networkpolicy.yaml
-	//│   ├── rolebinding.yaml
-	//│   └── role.yaml
-	//├── kustomization.yaml
-	//├── README.md
-	//├── tenant-a
-	//│   └── kustomization.yaml
-	//├── tenant-b
-	//│   └── kustomization.yaml
-	//├── tenant-c
-	//│   └── kustomization.yaml
-	//└── tenant-d
-	//    └── kustomization.yaml
-	// Root kustomization.yaml content:
-	// #kustomization.yaml
-	// resources:
-	// - tenant-a
-	// - tenant-b
-	// - tenant-c
-	// - tenant-d
-	// Base kustomization.yaml content:
-	// resources:
-	// - namespace.yaml
-	// - rolebinding.yaml
-	// - role.yaml
-	// - networkpolicy.yaml
-	// commonLabels:
-	//  test-case: hydration
-
-	// publicGCRImage pulls the public OCI image by the default `latest` tag
-	publicGCRImage = "gcr.io/stolos-dev/config-sync-ci/kustomize-components"
-	// publicARImage pulls the public OCI image by the default `latest` tag
-	publicARImage = "us-docker.pkg.dev/stolos-dev/config-sync-ci-public/kustomize-components"
-	// privateGCRImage pulls the private OCI image by digest
-	privateGCRImage = "us.gcr.io/stolos-dev/config-sync-ci/kustomize-components@sha256:e5b67f70a0cb3e7afb539521739baece7b8918282ccd6ef6be7caa061df95a3b"
-	// privateARImage pulls the private OCI image by tag
-	privateARImage    = "us-docker.pkg.dev/stolos-dev/config-sync-ci-private/kustomize-components:v1"
-	imageDigest       = "e5b67f70a0cb3e7afb539521739baece7b8918282ccd6ef6be7caa061df95a3b"
-	gsaARReaderEmail  = "e2e-test-ar-reader@stolos-dev.iam.gserviceaccount.com"
-	gsaGCRReaderEmail = "e2e-test-gcr-reader@stolos-dev.iam.gserviceaccount.com"
+	imageDigest = "e5b67f70a0cb3e7afb539521739baece7b8918282ccd6ef6be7caa061df95a3b"
 )
 
+// All the following images include a kustomization.yaml file in the root directory,
+// which includes resources for tenant-a, tenant-b, tenant-c, tenant-d.
+// Each tenant includes a NetworkPolicy, a Role and a RoleBinding.
+// Image structure:
+// .
+//├── base
+//│   ├── kustomization.yaml
+//│   ├── namespace.yaml
+//│   ├── networkpolicy.yaml
+//│   ├── rolebinding.yaml
+//│   └── role.yaml
+//├── kustomization.yaml
+//├── README.md
+//├── tenant-a
+//│   └── kustomization.yaml
+//├── tenant-b
+//│   └── kustomization.yaml
+//├── tenant-c
+//│   └── kustomization.yaml
+//└── tenant-d
+//    └── kustomization.yaml
+// Root kustomization.yaml content:
+// #kustomization.yaml
+// resources:
+// - tenant-a
+// - tenant-b
+// - tenant-c
+// - tenant-d
+// Base kustomization.yaml content:
+// resources:
+// - namespace.yaml
+// - rolebinding.yaml
+// - role.yaml
+// - networkpolicy.yaml
+// commonLabels:
+//  test-case: hydration
+
+// publicGCRImage pulls the public OCI image by the default `latest` tag
+func publicGCRImage() string {
```

Comment:
Are these images currently expected to be published manually outside of the test framework?

If so, maybe we can add some scripts to automate the publishing of these as a separate change.

Another alternative is that we use a non-floating tag and let the test scaffolding handle "find or create" style logic for the images.

I lean towards the latter

Reply:
What about publishing the images in the test? We want to test pulling via tag and digest. The digest is different for different projects.

Reply:
If we don't expect these to change very often then we could probably have some make targets that can be run manually to publish the images to the project used for testing. This adds some user overhead to make sure the image is published to the project before running the test.

I'm also good with the alternative of publishing the images as part of the test execution, although this may add some to the test execution time. An optimization could be that the scaffolding skips building/publishing the image if the tag is already published to the project's registry. 

I'm okay either way, although publishing in the test is probably more convenient for the developer

Reply:
The image content should not change. I like the idea of having a `make target` to publish the images to the testing project. It is a one-time change, so I think it is better than publishing in the test.

Reply:
SGTM. Feel free to unhold if you'd like to implement that as a separate change

Reply:
DONE. The OCI image digest is not fixed anymore. I updated the validation in the same PR to avoid test failures.

Reply:
I also created two new targets, one for pushing images to AR/GCR registries, and the other is for pushing changes to CSR repos.